### PR TITLE
Initial setup + unit tests

### PR DESCRIPTION
**Pull request by DSDubit**

### CODE REVIEW

File: Editor/LocalisationEditor.cs [hunk 1]
```diff
@@ -15,10 +14,11 @@ public class LocalisationEditor : EditorWindow
 		/// </summary>
 		public const int KEY_CRC_ENCODING_VERSION = 1;
 
+		// Code generation
```
Comment (FenrirWillow) [inline Editor/LocalisationEditor.cs]: I don't know what this comment means. Are these generated values? Are they related to some code generation?

File: Editor/LocalisationEditor.cs [hunk 2]
```diff
@@ -60,7 +60,7 @@ private void OnGUI()
 			if (localSchema == null)
 			{
 				EditorGUILayout.HelpBox(
-					"A key schema is a list of all the localisation keys in the project - generate a new one and populate it with all the keys you need, or drag the existing asset into the field above.",
+					"A key schema is a list of all the localisation keys in the project - generate a new one with the Assets menu and populate it with all the keys you need, or drag the existing asset into the field above.",
```
Comment (FenrirWillow) [inline Editor/LocalisationEditor.cs]: I think the grammatically correct is `using the Assets menu` :neckbeard:

File: Editor/LocalisationEditor.cs [hunk 3]
```diff
@@ -107,15 +110,15 @@ private void OnGUI()
...
 
-			EditorGUILayout.LabelField("Template required to generate locale class");
+			EditorGUILayout.LabelField("Template required to generate localisation class");
 			EditorGUILayout.BeginHorizontal();
 			configTemplate =
 				(TextAsset)EditorGUILayout.ObjectField("Config template:", configTemplate, typeof(TextAsset), false);
 
 			EditorGUI.BeginDisabledGroup(configTemplate == null);
-			if (GUILayout.Button("Generate Locale Class"))
+			if (GUILayout.Button("Generate Loc Class"))
```
Comment (FenrirWillow) [inline Editor/LocalisationEditor.cs]: What is a `Loc` class. I think the previous label was a little more descriptive.

File: LocalisationTable.cs [hunk 4]
```diff
@@ -15,18 +14,31 @@ public class LocalisationTable : ScriptableObject, ISerializationCallbackReceive
...
-		private List<int> keys = new List<int>();
+		private int crcEncodingVersion;
 
 		[SerializeField]
-		private List<string> values = new List<string>();
+		private List<int> keys = new List<int>();
 
 		[SerializeField]
-		public int crcEncodingVersion;
+		private List<string> values = new List<string>();
```
Comment (FenrirWillow) [inline LocalisationTable.cs:23]: This is a new private property but there is no new code that refers to it.

File: Editor/LocalisationEditor.cs (hunk 1 above)
Comment (DSDubit) [reply Editor/LocalisationEditor.cs]: Updated

File: Editor/LocalisationEditor.cs (hunk 2 above)
Comment (DSDubit) [reply Editor/LocalisationEditor.cs]: Modified

File: Editor/LocalisationEditor.cs (hunk 3 above)
Comment (DSDubit) [reply Editor/LocalisationEditor.cs]: Renamed - it's the shorthand utility/bridge class which exposes a 'get' method and enumerated keys, so callers don't have to call Localiser.GetLocalisedString(...) every time.

File: LocalisationTable.cs (hunk 4 above)
Comment (DSDubit) [reply LocalisationTable.cs:23]: I think all I did was re-order the fields, I don't see a new private property?

Comment (FenrirWillow) [reply LocalisationTable.cs:23]: Ah yes, they are just re ordered.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+Populate the schema with some meainingful data and keys. And make sure it's assigned in the Localisation window, as this is used to drive a lot of the rest of the localisation Editor UI. You only need one key schema per project, as it declares the keys the application knows about, not their values (yet).
+
+### Generate a Loc class
+
+This file is the 'bridge' the application will use to retrieve strings from the Localisation system. It declares all the enums the code will need to reference to do this, so for that reason, it needs to be automatically (re)generated every time you add or remove keys (or categories, or change category types, or anything else you edit in the schema). Once you're happy with your keys, assign the LocalisationConfigTemplate file in the 'Config template:' slot, and click 'Generate Loc Class'. This will autogenerate the class your application will need to reference to get strings - including the Keys enums and the Get(Enum key) function.
+
```
Comment (kkjamie) [inline README.md:32]: ` it needs to be automatically (re)generated every time you add or remove keys `

The wording here sounds like the system will automatically regenerate when you make a change? Is this so?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+Also, at the top of the table, you will see an array of 'Supported locales' - make sure this has at least one entry in (e.g. usually the default 'english' one has both en-GB and en-US) or it won't be used for any language.
+
+If you change anything in the schema, you will need to regenerate the Loc class (as above), but do NOT regenerate or overwrite the existing tables, or you will lose all the data you've already entered. Any keys you've added will appear in your table in the inspector ready to be populated with text, and any removed keys will disappear.
+
+## Application-side
+
+### Initialisation
+
+Refer to the API of DUCK.Localisation.Localiser.cs - but you will probably only need to do the following in most cases:
```
Comment (kkjamie) [inline README.md:45]: We could actually hyperlink this file in markdown to make it easier to open for reference. (Using relative paths in the repo)

**Review by LukeAvery92 — APPROVED** [review]

**Pull request merged**